include/cpp2util.h

```diff
@@ -959,577 +1250,459 @@ inline auto to_string(auto&& value, std::string_view) -> std::string
 //  TODO: Does this really warrant a new synonym? Perhaps "is void" is enough
 using empty = void;
 
+//  Type is Type
+//
+
+template <typename X, typename C>
+auto is() -> std::false_type { return {}; }
+
+template <typename X, typename C>
+    requires std::same_as<X, C> || std::derived_from<X,C>
+auto is() -> std::true_type { return {}; }
```

Comment:
I have found an issue:
```c++
auto x = is<T, U>();
x = !is<T, U>(); // error: no match for ‘operator=’ (operand types are ‘std::false_type’ {aka ‘std::integral_constant<bool, false>’} and ‘std::true_type’ {aka ‘std::integral_constant<bool, true>’})
```
So I think the return types need to be `bool`.

Reply:
@ntrel These functions are not intended to be used like that.
  
>`is` that is not using argument `x` for inspection returns `std::true_type` or `std::false_type`

This allows the compiler to eliminate at compile time possibilities that depend solely on the types involved, and not the runtime value of any objects.

You should never need to assign these values to variables, and if you do, you should never try to reassign one of those variables.

Reply:
An `is` expression should result in `bool`.
That means defining its result from the the call to a user-defined `operator is` as `bool(`_`call to user-defined `_`operator is)`.

Reply:
OK, thanks.